warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -286,105 +307,60 @@ public int compareTo(FieldSet o) {
         return combined;
     }
     
-    /**
-     * Detemine if two sets of values intersect
-     * 
-     * @param set1
-     * @param set2
-     * @return true if they have a value in common, false otherwise
-     */
+    // Return whether or not at least one element is found in both sets.
     private boolean intersects(Set<String> set1, Set<String> set2) {
-        for (String a : set1) {
-            if (set2.contains(a)) {
-                return true;
-            }
-        }
-        return false;
+        return set1.stream().anyMatch(set2::contains);
     }
     
-    /**
-     * Get the set of values for an attribute
-     * 
-     * @param attr
-     * @return the set of values
-     */
-    private Set<String> getValues(Attribute<?> attr) {
-        Set<String> values = new HashSet<>();
-        if (attr instanceof Attributes) {
-            for (Attribute<?> child : ((Attributes) attr).getAttributes()) {
-                values.addAll(getValues(child));
-            }
+    // Return the set of values for the provided attribute.
+    private Set<String> getValues(Attribute<?> attribute) {
+        if (attribute instanceof Attributes) {
+            // @formatter:off
+            return ((Attributes) attribute).getAttributes().stream()
+                            .map(this::getValues)
+                            .flatMap(Set::stream)
+                            .collect(Collectors.toSet());
+            // @formatter:on
         } else {
-            values.add(String.valueOf(attr.getData()));
+            return Collections.singleton(String.valueOf(attribute.getData()));
         }
-        return values;
     }
     
-    /**
-     * Get the base field name for a field (removed grouping context)
-     *
-     * @param documentField
-     * @return the base field name
-     */
-    private String getBaseFieldname(String documentField) {
-        int index = documentField.indexOf('.');
-        if (index < 0) {
-            return documentField;
-        } else {
-            return documentField.substring(0, index);
-        }
-    }
-    
-    /**
-     * Get the grouping context for a field
-     * 
-     * @param documentField
-     * @return the grouping context (first and last elements of the group). If no grouping context, then a unique group is given for this fieldname.
-     */
-    private String getGroupingContext(String documentField) {
-        String[] parts = StringUtils.split(documentField, '.');
+    // Return the grouping context for the provided field if it exists. If no grouping context is returned, then field.ungrouped is returned.
+    private String getGroupingContext(String field) {
+        String[] parts = StringUtils.split(field, '.');
```

Comment:
Consider swapping the datawave string utils for apache commons. Feel free to leave as-is.

Reply:
Sure, I don't have a particular attachment either way. Swapped for apache commons.